src/Filesystem/Dir.php

```diff
@@ -143,14 +143,18 @@ public static function files(
 	public static function index(
 		string $dir,
 		bool $recursive = false,
-		array|null $ignore = null,
+		array|null $ignore = [],
 		string $path = null
 	): array {
 		$result = [];
 		$dir    = realpath($dir);
 		$items  = static::read($dir);
 
 		foreach ($items as $item) {
+			if (in_array($item, $ignore) === true) {
```

Comment:
Needs to check if `$ignore` is an array first (could be `null`).

Reply:
Probably better not allow passing `null` - or what use case would passing `null` have?

Reply:
Not adding a breaking change. It could be that some users explicitly copied the default values into their method calls (for whatever reason). This will then break.

I'm in favor of removing the `null` value (or replacing it with `array|bool` like with `Dir::copy()` like I wrote above). But then it should happen in a major release.

Reply:
Let's change the type definition in the next major.